File: Assignment 2/Assn2.py
import sys
import re

file = sys.argv[1]
lines = [line.rstrip('\n') for line in open(file)]
pattern = re.compile("^[$][*]*([0]|([1-9][\\d]{2}|[1-9][\\d]|[1-9])([,][\\d]{3})*)([.][\\d]{2})?$")

for line in lines:
    if(pattern.match(line)):
        print "Matched: " + line
    else:
        print "Not Matched: " + line

File: Assignment 2/Assn2.js
var fs = require('fs');
var readline = require('readline');
var instream = fs.createReadStream(process.argv[2]);
var outstream = new (require('stream'))();
var rl = readline.createInterface(instream,outstream);
var regex = '^[$][*]*'
+ '([0]|([1-9][\\d]{2}|[1-9][\\d]|[1-9])([,][\\d]{3})*)'
+ '([.][\\d]{2})?$';

rl.on('line', function(line) {
  if(line.match(regex)) {
    console.log('Matched: ' + line)
  }
  else {
    console.log("Not matched: " + line)
  }
});
rl.on('close', function(line) {
  
})
